main.js

```diff
@@ -0,0 +1,280 @@
+'use strict'
+var DEBUG = false
+if(DEBUG){
+  var dbg = document.createElement('p')
+  dbg.textContent = 'default text'
+  dbg.style.position = 'absolute'
+  dbg.style.top = '0px'
+  document.querySelector('body').appendChild(dbg)
+}
+
+function dbgt() {
+  var argumentString = ''
+  for(var i of arguments) {
+    argumentString += i
+  }
+  dbg.textContent = '$ '+argumentString
+  console.log(argumentString)
+}
+
+(function(){
+  var originalErrFunction = console.error
+
+  console.error = function() {
+    var debugMessage = ['Error: ']
+    for(var i of arguments) {
+      debugMessage.push(i)
+    }
+    dbgt(debugMessage.join(''))
+    originalErrFunction.apply(null, arguments)
+  }
+}())
+
+function Calculator() {
+  this.operands = []
+  this.tailIndex = 0
+  this.inputSize = 3.0
+  this.clearOrAllClear = 'all'
+  this.domElements = {
+    'Numbers': [
+      document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(1) > button'),
```

Comment:
So, I know that what they were looking for when designing this project was to have a separation of concerns between the DOM and the calculator.  If your html elements change at all, this will be a lot of work to alter (for example, if you get to step 3 and are adding the second calculator.)

They way they want you to do this is to tack an event listener onto your calculator container.  Then, in the callback function for that event listener, you can check `event.target` to see which button was clicked.  Then you can use that to call functions inside your calculator.  The calculator itself doesn't need to know anything about the DOM.